### PR TITLE
Gapic fixes

### PR DESCRIPTION
**Pull request by ben-clayton**

### CODE REVIEW

File: gapic/src/main/com/google/gapid/widgets/ImagePanel.java [hunk 1]
```diff
@@ -130,10 +130,16 @@
   private final StatusBar status;
   protected final ImageComponent imageComponent;
   private final BackgroundSelection backgroundSelection;
-  private ToolItem zoomFitItem, backgroundItem, saveItem, colorChanelsItem;
+  private ToolItem zoomFitItem, zoomActualItem, backgroundItem, saveItem, colorChanelsItem;
   private MultiLayerAndLevelImage image = MultiLayerAndLevelImage.EMPTY;
   private Image[] layers = NO_LAYERS;
 
+  public enum ZoomMode {
```
Comment (pmuetschard) [inline gapic/src/main/com/google/gapid/widgets/ImagePanel.java]: private?

File: gapic/src/main/com/google/gapid/widgets/ImagePanel.java [hunk 2]
```diff
@@ -885,17 +888,20 @@ public Pixel getPixel(Point point) {
...
-        refresh();
-      }
+    public void setZoomMode(ZoomMode mode) {
+      this.zoomMode = mode;
+      updateZoomFromMode();
     }
 
-    public void zoomToActual() {
-      setScale(DPIUtil.autoScaleDown(1.0f));
+    public void updateZoomFromMode() {
```
Comment (pmuetschard) [inline gapic/src/main/com/google/gapid/widgets/ImagePanel.java]: private?

File: gapic/src/main/com/google/gapid/widgets/ImagePanel.java [hunk 3]
```diff
@@ -885,17 +888,20 @@ public Pixel getPixel(Point point) {
...
+      switch (this.zoomMode) {
```
Comment (pmuetschard) [inline gapic/src/main/com/google/gapid/widgets/ImagePanel.java]: nit: no "this"

File: gapic/src/main/com/google/gapid/widgets/ImagePanel.java [hunk 4]
```diff
@@ -1088,9 +1088,12 @@ public void init(Renderer renderer) {
     public void update(Renderer renderer, SceneData newData) {
       // Release textures that are no longer in data.
       Set<Image> newSet = Sets.newHashSet(newData.images);
-      for (Image removed : Sets.difference(imageToTexture.keySet(), newSet)) {
-        imageToTexture.remove(removed).delete();
+      Set<Image> toRemove = Sets.difference(imageToTexture.keySet(), newSet);
```
Comment (pmuetschard) [inline gapic/src/main/com/google/gapid/widgets/ImagePanel.java]: I wonder how many more bugs we can spawn at this location.... :disappointed:

**Comment by ben-clayton** [comment]
I'm out tonight.
Are you happy if I submit this as is? I want to get this in in case we chose to spin another build tonight.

**Comment by pmuetschard** [comment]
submitted.

**Pull request merged**